src/messagingsession/DefaultMessagingSession.ts

```diff
@@ -224,24 +236,39 @@ export default class DefaultMessagingSession implements MessagingSession {
             observer.messagingSessionDidStart();
           }
         });
+        this.bootstrapResolved();
         this.isSessionEstablished = true;
+
+        // Send message and flush the queue.
+        const preBootstrapMessageLength = this.preBootstrapMessages.length;
+        for (let iter = 0; iter < preBootstrapMessageLength; iter++) {
+          const preBootstrapMessage = this.preBootstrapMessages.shift();
+          this.forEachObserver(observer => {
+            this.sendMessageToObserver(observer, preBootstrapMessage);
+          });
+        }
       } else if (!this.isSessionEstablished) {
         // SESSION_ESTABLISHED is not guaranteed to be the first message, and in rare conditions a message or event from
         // a channel the member is a member of might arrive prior to SESSION_ESTABLISHED.  Because SESSION_ESTABLISHED indicates
-        // it is safe to bootstrap the user application with out any race conditions in losing events we opt to drop messages prior
-        // to SESSION_ESTABLISHED being received
+        // it is safe to bootstrap the user application without any race conditions in losing events we opt to store messages prior
+        // to SESSION_ESTABLISHED being received and send when once SESSION_ESTABLISHED.
+        this.preBootstrapMessages.push(message);
```

Comment:
On average  how many messages would be in this queue and how often this happens? Do we need to limit the size of this queue?

Reply:
We believe this to be a rare race condition after introducing prefetch feature. Maximum number of messages we sent via Prefetch is 50.